src/ape/contextmanagers.py

```diff
@@ -0,0 +1,17 @@
+import contextlib
+from typing import List, Optional
+
+from ape_accounts.accounts import AccountAPI
+
+_DEFAULT_SENDERS: List[Optional[AccountAPI]] = []
+
+
+@contextlib.contextmanager
+def use_sender(address):
```

Comment:
Thoughts on making this off the `AccountManager` and `TestAccountManager` such that the syntax could be like:

```python
# myscript.py
from ape import accounts

from .contract import my_contract  # Made up

def main():
    with account.use_sender("joebob"):  # Also allow address?
        my_contract.mutableMethod(123)
```

and:

```python
# test_mycontract.py
def test_mymutablemethod(accounts, my_contract):
    with account.use_sender(0):  # where 0 is the first test account
        my_contract.mutableMethod(123)
```

Reply:
that. address is actually directly an account. AccountAPI

Reply:
```python
with accounts.use_sender(account_api)
```

Reply:
I like using `accounts` as the way to use this context manager, we can make it be abstract such that you can use `Union[AccountAPI, AddressType, ENS, Alias]`:
1. use `AccountAPI` directly (like it does now), for both `AccountManager` and `TestAccountManager`
2. `accounts.load` if `AccountManager` and `Alias` (scripting usage)
3. `convert(address_or_ens, AddressType` then `accounts[address]` if anything else provided, for both `AccountManager` and `TestAccountManager`

Also could expand scripting usage with `@account_option` click decorator setting this up too, so that scripts don't even have to use `sender=` at all (not sure about this, I sort of like the explicitness a bit)

Reply:
changed my implementation a bit @fubuloubu @unparalleled-js